addons/website_sale/models/sale_order.py

```diff
@@ -101,224 +99,165 @@ def _cart_update(self, cr, uid, ids, product_id=None, line_id=None, add_qty=0, s
+    def _compute_pricelist_id(self):
+        for website in self:
+            website.pricelist_id = website.with_context(website_id=website.id).get_current_pricelist()
 
+    @tools.ormcache('self.env.uid', 'country_code', 'show_visible', 'website_pl', 'current_pl', 'all_pl')
+    def _get_pl(self, country_code, show_visible, website_pl, current_pl, all_pl):
+        """ Return the list of pricelists that can be used on website for the current user.
         :param str country_code: code iso or False, If set, we search only price list available for this country
         :param bool show_visible: if True, we don't display pricelist where selectable is False (Eg: Code promo)
         :param int website_pl: The default pricelist used on this website
         :param int current_pl: The current pricelist used on the website
                                (If not selectable but the current pricelist we had this pricelist anyway)
         :param list all_pl: List of all pricelist available for this website
-
         :returns: list of pricelist ids
         """
-        pcs = []
+        pricelists = self.env['product.pricelist']
 
         if country_code:
-            groups = self.pool['res.country.group'].search(cr, uid, [('country_ids.code', '=', country_code)])
-            for cgroup in self.pool['res.country.group'].browse(cr, uid, groups):
-                for pll in cgroup.website_pricelist_ids:
-                    if not show_visible or pll.selectable or pll.pricelist_id.id == current_pl:
-                        pcs.append(pll.pricelist_id)
-
-        if not pcs:  # no pricelist for this country, or no GeoIP
-            pcs = [pll.pricelist_id for pll in all_pl
-                   if not show_visible or pll.selectable or pll.pricelist_id.id == current_pl]
-
-        partner = self.pool['res.users'].browse(cr, SUPERUSER_ID, uid).partner_id
-        if not pcs or partner.property_product_pricelist.id != website_pl:
-            pcs.append(partner.property_product_pricelist)
-        # remove duplicates and sort by name
-        pcs = sorted(set(pcs), key=lambda pl: pl.name)
-        return [pl.id for pl in pcs]
-
-    def get_pricelist_available(self, cr, uid, show_visible=False, context=None):
+            for cgroup in self.env['res.country.group'].search([('country_ids.code', '=', country_code)]):
+                pricelists += cgroup.website_pricelist_ids.filtered(lambda pl: not show_visible or pl.selectable or pl.pricelist_id.id == current_pl).mapped('pricelist_id')
+
+        if not pricelists:  # no pricelist for this country, or no GeoIP
+            pricelists += all_pl.filtered(lambda pl: not show_visible or pl.selectable or pl.pricelist_id.id == current_pl).mapped('pricelist_id')
+
+        partner = self.env.user.partner_id
+        if not pricelists or partner.property_product_pricelist.id != website_pl:
+            pricelists += partner.property_product_pricelist
+
+        return pricelists.sorted(lambda pl: pl.name)
```

Comment:
@rga-odoo did you remove the comment because you noticed it doesn't remove duplicates anymore?